Sources/SwiftProtobuf/Message+BinaryAdditions.swift

```diff
@@ -57,31 +57,6 @@ extension Message {
     return visitor.serializedSize
   }
 
-  /// Creates a new message by decoding the given `Data` value containing a
-  /// serialized message in Protocol Buffer binary format.
-  ///
-  /// - Parameters:
-  ///   - serializedData: The binary-encoded message data to decode.
-  ///   - extensions: An `ExtensionMap` used to look up and decode any
-  ///     extensions in this message or messages nested within this message's
-  ///     fields.
-  ///   - partial: If `false` (the default), this method will check
-  ///     `Message.isInitialized` before encoding to verify that all required
-  ///     fields are present. If any are missing, this method throws
-  ///     `BinaryEncodingError.missingRequiredFields`.
-  ///   - options: The BinaryDecodingOptions to use.
-  /// - Throws: `BinaryDecodingError` if decoding fails.
-  @inlinable
-  public init(
-    serializedData data: Data,
```

Comment:
I wonder if we should keep the old names to help folks in migration.

Yes, we're in a 2.0 phase, but this sorta breaking change could be really hard for teams with large code bases and multiple dependencies that all use SwiftProtobuf. Seems like allowing those sub dependencies to migrate one at a time would be worth while.

Reply:
IMO we should provide this method in a new SwiftProtobufFoundationCompat module. This should ease migration

Reply:
Sounds like there some plan being worked on since I haven't heard "SwiftProtobufFoundationCompat" 😄 , maybe doing a bit of a outline on the expected path forward on the issue would help rather then PRs that are only partial pictures into a plan.

Reply:
I'll add this this method (i.e. with the old signature) to the FoundationCompat module, and have it call the new method.